app/src/main/java/org/kiwix/kiwixmobile/main/NavigationHostFragment.kt

```diff
@@ -0,0 +1,105 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.main
+
+import android.content.Intent
+import android.view.ActionMode
+import android.view.Menu
+import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.fragment.NavHostFragment
+import org.kiwix.kiwixmobile.core.base.BaseFragmentActivityExtensions
+import org.kiwix.kiwixmobile.core.base.BaseFragmentActivityExtensions.Super
+import org.kiwix.kiwixmobile.core.base.BaseFragmentActivityExtensions.Super.ShouldCall
+import org.kiwix.kiwixmobile.core.main.KiwixWebView
+import org.kiwix.kiwixmobile.core.main.WebViewProvider
+
+class NavigationHostFragment : NavHostFragment(), WebViewProvider, BaseFragmentActivityExtensions {
+  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    super.onActivityResult(requestCode, resultCode, data)
+    childFragmentManager.fragments.forEach { it.onActivityResult(requestCode, resultCode, data) }
+  }
+
+  override fun getCurrentWebView(): KiwixWebView? {
+    return childFragmentManager.fragments.filterIsInstance<WebViewProvider>().firstOrNull()
+      ?.getCurrentWebView()
+  }
+
+  override fun onRequestPermissionsResult(
+    requestCode: Int,
+    permissions: Array<out String>,
+    grantResults: IntArray
+  ) {
+    super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    childFragmentManager.fragments.forEach {
+      it.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    }
+  }
+
+  override fun onActionModeStarted(
+    actionMode: ActionMode,
+    activity: AppCompatActivity
+  ): Super {
+    var result = ShouldCall
+    childFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
+      result = it.onActionModeStarted(actionMode, activity)
+    }
+    return result
+  }
+
+  override fun onActionModeFinished(
+    actionMode: ActionMode,
+    activity: AppCompatActivity
+  ): Super {
+    var result = ShouldCall
+    childFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
+      result = it.onActionModeFinished(actionMode, activity)
+    }
+    return result
+  }
+
+  override fun onBackPressed(activity: AppCompatActivity): Super {
+    var result = ShouldCall
+    childFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
+      result = it.onBackPressed(activity)
+    }
+    return result
+  }
+
+  override fun onNewIntent(
+    intent: Intent,
+    activity: AppCompatActivity
+  ): Super {
+    var result = ShouldCall
+    childFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
+      result = it.onNewIntent(intent, activity)
```

Comment:
so this only stores the last result, should there be a type of and-ing here? Or is expected behaviour that previous results are ignored?

Reply:
ooh, good question. I am thinking it should only call super if all children agree on it calling super. (and-ing here)